### PR TITLE
Some region table refactoring

### PR DESCRIPTION
**Pull request by pb-cdunn**
Unittests still pass. They begin to fail at the *next* commit on branch *bam_concordant*, so this is a good check-point for review.

### CODE REVIEW

File: pbdata/reads/RegionAnnotation.hpp [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+inline
+bool compare_region_annotation_by_type(const RegionAnnotation & lhs,
+                                       const RegionAnnotation & rhs)
+{
+    if (lhs.GetHoleNumber() == rhs.GetHoleNumber()) {
+        if (lhs.GetTypeIndex() == rhs.GetTypeIndex()) {
+            if (lhs.GetStart() == rhs.GetStart()) {
+                 if (lhs.GetEnd() == rhs.GetEnd())
+                     return lhs.GetScore() < rhs.GetScore();
+                 else return lhs.GetEnd() > rhs.GetEnd();
```
Comment (pb-cdunn) [inline pbdata/reads/RegionAnnotation.hpp]: `GetEnd` has the reverse sense of the other comparisons. Are you 100% certain of correctness?

File: unittest/pbdata/reads/RegionAnnotations_gtest.cpp [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+ *        Created:  09/27/2015 03:54:55 PM
+ *       Compiler:  gcc
+ *
+ *         Author:  Yuan Li (yli), yli@pacificbiosciences.com
+ *        Company:  Pacific Biosciences
+ *
+ * ==================================================================
+ */
+#include "gtest/gtest.h"
+#define private public
```
Comment (pb-cdunn) [inline unittest/pbdata/reads/RegionAnnotations_gtest.cpp]: :neutral_face:

Oh, well.

File: unittest/pbdata/reads/RegionAnnotations_gtest.cpp [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+static const std::vector<RegionAnnotation> REGIONS_SORTED_BY_TYPE = {
+    RegionAnnotation(HOLENUMBER, 0, 249, 329, 800),// adapter
+    RegionAnnotation(HOLENUMBER, 0, 649, 700, 700),// adapter
+    RegionAnnotation(HOLENUMBER, 1, 0,   250, -1), // insert
+    RegionAnnotation(HOLENUMBER, 1, 300, 650, -1), // insert
+    RegionAnnotation(HOLENUMBER, 1, 700, 999, -1), // insert
+    RegionAnnotation(HOLENUMBER, 2, 50,  900, 900) // hqregion
+};
+
+/*
```
Comment (pb-cdunn) [inline unittest/pbdata/reads/RegionAnnotations_gtest.cpp]: I think I saw this uncommented in a future commit, so we can ignore this for now.

**Comment by pb-cdunn** [comment]
Looks fine to me. Unless there are objections, I'll merge this part soon.

File: pbdata/reads/RegionAnnotation.hpp (hunk 1 above)
Comment (ylipacbio) [inline pbdata/reads/RegionAnnotation.hpp]: yes, GetEnd has reverse comparison.
Although this is not likely to run given our existing region table, in which regions of the same type should not intersect with each other, it is my personal preference that a 'super' interval should be placed in front of all its 'subset' intervals after sort (.e.g., after sort, [0, 1000] be placed in front of [0, 100])

**Pull request merged**